homeassistant/components/pi_hole/config_flow.py

```diff
@@ -55,27 +61,35 @@ async def async_step_init(self, user_input, is_import=False):
             location = user_input[CONF_LOCATION]
             tls = user_input[CONF_SSL]
             verify_tls = user_input[CONF_VERIFY_SSL]
-            api_token = user_input.get(CONF_API_KEY)
             endpoint = f"{host}/{location}"
 
             if await self._async_endpoint_existed(endpoint):
                 return self.async_abort(reason="already_configured")
 
             try:
-                await self._async_try_connect(
-                    host, location, tls, verify_tls, api_token
-                )
-                return self.async_create_entry(
-                    title=name,
-                    data={
-                        CONF_HOST: host,
-                        CONF_NAME: name,
-                        CONF_LOCATION: location,
-                        CONF_SSL: tls,
-                        CONF_VERIFY_SSL: verify_tls,
-                        CONF_API_KEY: api_token,
-                    },
-                )
+                await self._async_try_connect(host, location, tls, verify_tls)
+                self._config = {
+                    CONF_HOST: host,
+                    CONF_NAME: name,
+                    CONF_LOCATION: location,
+                    CONF_SSL: tls,
+                    CONF_VERIFY_SSL: verify_tls,
+                }
+                if is_import:
+                    api_key = user_input.get(CONF_API_KEY)
+                    return self.async_create_entry(
+                        title=name,
+                        data={
+                            **self._config,
+                            CONF_STATISTICS_ONLY: api_key is None,
+                            CONF_API_KEY: api_key,
```

Comment:
Since one value is derived from the other, why store both?

Reply:
For config entries imported from the YAML configuration, not providing API key means statistics only. However, for config entries created through UI, this is not always the case. We store both values here so that the config entries always have the same structure, which makes setting up the entries easier.